src/InteractiveWindow/Editor/History.cs

```diff
@@ -149,16 +149,7 @@ internal void Add(SnapshotSpan span)
 
         internal Entry GetNext(string pattern)
         {
-            var startPos = _current;
-            Entry next;
-            next = MoveNext(pattern);
-            if (next == null)
-            {
-                _current = startPos;
-                return null;
-            }
-
-            return next;
+            return MoveNext(pattern);
```

Comment:
@ManishJayaswal I don't believe so.  The tests seem to indicate that it works as expected.  (Note that previous and next are not symmetrical because of the (possible) presence of uncommitted user input after next.)